src/main/java/appeng/recipes/factories/recipes/CuttingRecipeFactory.java

```diff
@@ -0,0 +1,162 @@
+
+package appeng.recipes.factories.recipes;
+
+
+import java.util.Iterator;
+
+import javax.annotation.Nonnull;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import com.google.gson.JsonSyntaxException;
+
+import net.minecraft.inventory.InventoryCrafting;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.crafting.IRecipe;
+import net.minecraft.item.crafting.Ingredient;
+import net.minecraft.util.JsonUtils;
+import net.minecraft.util.NonNullList;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.world.World;
+import net.minecraftforge.common.crafting.CraftingHelper;
+import net.minecraftforge.common.crafting.IRecipeFactory;
+import net.minecraftforge.common.crafting.JsonContext;
+import net.minecraftforge.oredict.ShapelessOreRecipe;
+
+import appeng.api.recipes.ResolverResult;
+import appeng.core.AELog;
+import appeng.core.Api;
+import appeng.core.AppEng;
+
+
+public class CuttingRecipeFactory implements IRecipeFactory
+{
+	@Override
+	public IRecipe parse( JsonContext context, JsonObject json )
+	{
+		String group = JsonUtils.getString( json, "group", "" );
+
+		NonNullList<Ingredient> ings = NonNullList.create();
+		for( JsonElement ele : JsonUtils.getJsonArray( json, "ingredients" ) )
+		{
+			ings.add( CraftingHelper.getIngredient( ele, context ) );
+		}
+
+		if( ings.isEmpty() )
+		{
+			throw new JsonParseException( "No ingredients for shapeless recipe" );
+		}
+
+		JsonObject resultObject = (JsonObject) json.get( "result" );
+		int count = JsonUtils.getInt( resultObject, "count", 1 );
+
+		ItemStack itemStack = ItemStack.EMPTY;
+		if( resultObject.has( "part" ) )
+		{
+			String ingredient = resultObject.get( "part" ).getAsString();
+			Object result = (Object) Api.INSTANCE.registries().recipes().resolveItem( AppEng.MOD_ID, ingredient );
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unnecessary cast to "Object". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1905)